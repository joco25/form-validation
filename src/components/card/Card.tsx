import React,{FC} from "react"
import { ICourseDetails } from "../inputs/helpers/course.interface";
import "./styles/Card.css"

export interface CardProps extends ICourseDetails {}

const Card:FC<CardProps>= ({department, courseNumber, year, semester}) => {
  return (
    <div className="card">
        <div className="card__title">{department} {courseNumber}</div>
        <div className="card__container">
            <div className="row">
                <span className="column">Department:</span> 
                <span className="column">{department}</span>
            </div>
            <div className="row">
                <span className="column">Course:</span> 
                <span className="column">{courseNumber}</span>
            </div>
            <div className="row">
                <span className="column">Year:</span> 
                <span className="column">{year}</span>
            </div>
            <div className="row">
                <span className="column">Semester:</span> 
                <span className="column">{semester}</span>
            </div>
        </div>
    </div>
  )
}

export default Card;