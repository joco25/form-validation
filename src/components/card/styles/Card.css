.card {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-top: 20px;
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.1);
  border-radius: 3px;
  overflow: hidden;
}

.card__title {
  display: flex;
  flex: 1;
  width: 100%;
  background-color: blue;
  color: white;
  font-size: 1.2rem;
  padding: 5px 10px 5px 10px;
}

.card__container {
  padding: 10px;
  width: 100%;
  box-sizing: border-box;
}

.row {
  display: flex;
  flex-direction: row;
  width: 100%;
  margin-bottom: 3px;
}

.column {
  flex: 50%;
}
