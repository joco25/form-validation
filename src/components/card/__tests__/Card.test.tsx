import React from 'react';
import { render } from '@testing-library/react';
import Card, { CardProps } from '../Card';

describe('Card', () => {
  const defaultProps: CardProps = {
    department: 'Computer Science',
    courseNumber: 101,
    year: 2022,
    semester: 'Spring',
  };
  
  it('should match snapshot', () => {
    const { container } = render(<Card {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
