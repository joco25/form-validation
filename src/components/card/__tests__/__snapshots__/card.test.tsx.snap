// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card__title"
    >
      Computer Science
       
      101
    </div>
    <div
      class="card__container"
    >
      <div
        class="row"
      >
        <span
          class="column"
        >
          Department:
        </span>
        <span
          class="column"
        >
          Computer Science
        </span>
      </div>
      <div
        class="row"
      >
        <span
          class="column"
        >
          Course:
        </span>
        <span
          class="column"
        >
          101
        </span>
      </div>
      <div
        class="row"
      >
        <span
          class="column"
        >
          Year:
        </span>
        <span
          class="column"
        >
          2022
        </span>
      </div>
      <div
        class="row"
      >
        <span
          class="column"
        >
          Semester:
        </span>
        <span
          class="column"
        >
          Spring
        </span>
      </div>
    </div>
  </div>
</div>
`;
